core/src/main/scala/cats/data/Ior.scala

```diff
@@ -947,12 +952,27 @@ sealed abstract private[data] class IorInstances0 {
     new Traverse[Ior[A, *]] {
       def traverse[F[_]: Applicative, B, C](fa: A Ior B)(f: B => F[C]): F[A Ior C] =
         fa.traverse(f)
+
+      override def mapAccumulate[S, B, C](init: S, fa: Ior[A, B])(f: (S, B) => (S, C)): (S, Ior[A, C]) =
+        fa match {
+          case l @ Ior.Left(_) => (init, l)
+          case Ior.Right(b) =>
+            val (snext, c) = f(init, b)
+            (snext, Ior.Right(c))
+          case Ior.Both(a, b) =>
+            val (snext, c) = f(init, b)
+            (snext, Ior.Both(a, c))
+        }
       def foldLeft[B, C](fa: A Ior B, b: C)(f: (C, B) => C): C =
         fa.foldLeft(b)(f)
       def foldRight[B, C](fa: A Ior B, lc: Eval[C])(f: (B, Eval[C]) => Eval[C]): Eval[C] =
         fa.foldRight(lc)(f)
 
-      override def size[B](fa: A Ior B): Long = fa.fold(_ => 0L, _ => 1L, (_, _) => 1L)
+      override def size[B](fa: A Ior B): Long =
```

Comment:
optimization to avoid lambdas when pattern matching will do.

Reply:
How about simply `if (fa.isLeft) 0L else 1L`.

Reply:
Oh, yikes, `isLeft` and friends need to be optimized too (maybe by dynamic dispatch which is how `Option` is implemented?). Regardless, I still think this method would be better written in terms of `isLeft`.

Reply:
done.